### PR TITLE
Follow-on work to goldens

### PR DESCRIPTION
**Pull request by tvolkert**
* Exclude flutter_goldens package from dartdoc because it's for internal
  use only
* Document why flutter_tools doesn' tneed to be excluded from the list of
  packages to document
* Make `flutter analyze --dartdocs` ignore undocumented members in flutter_goldens
* Performance optimization in the flutter comparator, and associated
  test updates.

Fixes the undocumented public members benchmark

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/analyze_once.dart [hunk 1]
```diff
@@ -273,6 +273,8 @@ class AnalyzeOnce extends AnalyzeBase {
       return false;
     if (filenameComponents[flutterRootComponents.length + 1] == 'flutter_tools')
       return false;
+    if (filenameComponents[flutterRootComponents.length + 1] == 'flutter_goldens')
+      return false;
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/analyze_once.dart]: fyi @devoncarew

**Review by cbracken — APPROVED** [review]

**Comment by Hixie** [comment]
LGTM

**Comment by cbracken** [comment]
lgtm

File: packages/flutter_tools/lib/src/commands/analyze_once.dart (hunk 1 above)
Comment (devoncarew) [reply packages/flutter_tools/lib/src/commands/analyze_once.dart]: :grimacing: oh no, my giant PR

**Pull request merged**